src/components/eventList/EventList.tsx

```diff
@@ -0,0 +1,130 @@
+import {useEffect, useState} from 'react';
+import {FlatList, Text, TextInput, TouchableOpacity, View} from 'react-native';
+import Icon from 'react-native-vector-icons/Feather';
+
+import {getData} from '@storage/storage';
+import {storageKeys} from '@constants/storageKeys';
+import {useUserLoginContext} from '@contexts/LoginContext';
+import EventFilter from '@components/eventFilter/EventFilter';
+import EventCard from '@components/eventCard/EventCard';
+import CustomModal from '@components/customModal/CustomModal';
+import {sortData} from '@constants/constant';
+import {searchEvents, sortEvents} from '@helpers/helper';
+
+import {colors} from '@theme/ColorStyles';
+import {styles} from './styles';
+
+const EventList = () => {
+
+  const {loginId} = useUserLoginContext();
+  const [searchTerm, setSearchTerm] = useState('');
+  const [eventsArr, setEventsArr] = useState<Events>([]);
+  const [summarizedEventsArr, setSummarizedEventsArr] = useState<Events>([]);
+  const [sortVisible, setSortVisible] = useState(false);
+  const [filterVisible, setFilterVisible] = useState(false);
+  const [sortSelected, setSortSelected] = useState<DropdownProps>({
+    label: 'Date',
+    value: 'date',
+  });
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      const userData = await getData(`${storageKeys.eventData}-${loginId}`);
+      setEventsArr(userData);
+      setSummarizedEventsArr(userData);
+    };
+
+    fetchUsers();
+  }, []);
+
+  const sortedEvents = sortEvents(summarizedEventsArr, sortSelected);
```

Comment:
`sortedEvents` use memoised value

`SearchedEvent` use small `s` and memoised value